File: Data Structures & Algorithms/permutations/submission-0.cpp
class Solution {
public:
   void solve(vector<int>& nums, vector<vector<int>>& res, 
   vector<int>&ds, int n, unordered_set<int>& st) {

        if(ds.size() >= n) {
            res.push_back(ds);
            return ;
        }

        for(int i = 0; i < n; i++) {
            if(st.find(nums[i]) == st.end()) {
                ds.push_back(nums[i]);
                st.insert(nums[i]);
                solve(nums, res, ds, n, st);
                ds.pop_back();
                st.erase(nums[i]);
            }
        }

   }
    vector<vector<int>> permute(vector<int>& nums) {
        vector<vector<int>> res; 
        int n = nums.size();
        vector<int> ds;
        unordered_set<int> st;

        solve(nums, res, ds, n, st);
        
        
        return res;
    }
};

File: Data Structures & Algorithms/concatenation-of-array/submission-0.java
class Solution {
    public int[] getConcatenation(int[] nums) {
        int n = nums.length;
        int newSize = 2*n, i = 0, k = 0;
        int[] ans = new int[newSize];
        for(i = 0; i < n; i++) {
            ans[i] = nums[i];
        }
        for(int j = i; j < newSize; j++){
            ans[j] = nums[k++];
        }
        return ans;
    }
}